cohorts/dataframe_loader.py

```diff
@@ -29,7 +29,9 @@ class DataFrameLoader(object):
     def __init__(self,
                  name,
                  load_dataframe,
-                 join_on="patient_id"):
+                 join_on="patient_id",
```

Comment:
Rename this to `join_on_right`? Otherwise I think it's confusing.

If you're worried about backwards compat., can add an additional `join_on` parameter defaulted to `None`; if passed in, log a https://docs.python.org/3/library/exceptions.html#DeprecationWarning and pass to `join_on_right`?